### ISSUE TITLE
neo4j-admin import fails to parse csv

### ISSUE THREAD

**Issue by tolomaus** [neo4j/neo4j#11218]
- Neo4j version: 3.3.3-1 installed through yum
- Operating system: centos 7
- API/Driver: neo4j-admin import

I'm loading a large (1M+ records with 40 fields) csv file with the following (simplified) command:
neo4j-admin import \
    --nodes:Organization /path/to/organizations.csv \
    --multiline-fields=true \
    --ignore-duplicate-nodes=true \
    --ignore-missing-nodes=true \
    --array-delimiter ";"

And since a couple of weeks it regularly fails with: 
```
WARNING Import failed. The store files in /path/to/graph.db are left as they are, although they are likely in an unusable state. Starting a database on these store files will likely fail or observe inconsistent records so start at your own risk or delete the store manually
unexpected error: ERROR in input
  data source: BufferedCharSeeker[source:/path/to/organizations.csv, position:301989889, line:1100684]
  in field: abbreviationFr:string:6
  for header: [vat:ID(Organization), ... , abbreviationFr:string, ...]
  raw field value: ??
  original error: String index out of range: -1
```

This error occurs deterministically in the sense that once it happens it will consistently happen with the same input. However, it may just go away after simply reordering the fields, removing a line or sometimes even a cell. I have tried to isolate the issue by only taking the lines around the line number mentioned in the error as the input but I havent been able to reproduce the error in this way.

The csv file itself should be valid csv since it was generated from Spark.

Any idea what could be wrong? Did you make any changes to the import in the recent versions? Anything I can do to pinpoint the issue?

Thanks in advance,

Niek.

**Comment by tolomaus** [comment]
downgraded to 3.3.1 and all seems to be working fine so far

**Comment by tinwelint** [comment]
Hello @tolomaus have you got a sample CSV file or means of generating one which produces this issue?

**Comment by tolomaus** [comment]
Hi @tinwelint ,

Unfortunately no. I tried to pinpoint the issue with a small subset of the data but could never reproduce the issue.

I could send you the complete 100+MB file if that helps.

**Comment by tinwelint** [comment]
Alright, I think it's alright... I've most likely found the issue (see linked PR).

**Comment by tolomaus** [comment]
Cool! Would you like me to test your fix on my dataset? I could find some time later this week, just let me know.

**Comment by tinwelint** [comment]
That would be absolutely brilliant, Niek. You'll probably get pinged when the PR is merged. Also I have to merge it forward to 3.3 and 3.4 so I can ping you when all is done.

**Comment by tinwelint** [comment]
@tolomaus this has now been merged into 3.2, 3.3 and 3.4 version branches of neo4j. Maintenance releases of these versions will come later, don't know exactly when right now. But feel free to try this fix out on the 3.3 version branch if you want to.
